src/services/infra/InfraService.ts

```diff
@@ -91,4 +105,174 @@ export default class InfraService {
+       * does not contain the DNS redirections info.
+       */
+
+      const dnsInfo = await this.launchesService.getDomainAssociationRecord(
+        primaryDomain,
+        appId
+      )
+
+      const certificationRecord = this.parseDNSRecords(
+        dnsInfo.domainAssociation?.certificateVerificationDNSRecord
+      )
+      if (!certificationRecord) {
+        throw new Error(`error while parsing ${dnsInfo}`)
+      }
+
+      const {
+        source: domainValidationSource,
+        target: domainValidationTarget,
+      } = certificationRecord
+
+      const subDomainList = dnsInfo.domainAssociation?.subDomains
+      if (!subDomainList || !subDomainList[0].dnsRecord) {
+        throw Error("subdomain list not created yet")
+      }
+
+      const primaryDomainInfo = this.parseDNSRecords(subDomainList[0].dnsRecord)
+
+      if (!primaryDomainInfo) {
+        throw Error("primary domain info not created yet")
+      }
+
+      /**
+       * shape of dnsInfo.domainAssociation.subDomains:
+       * {
+       *   dnsRecord: "CNAME gibberish.cloudfront.net",
+       *   subDomainSettings: {
+       *     branchName : "master",
+       *     prefix? : "www"
+       *   }
+       * }
+       */
+
+      const primaryDomainTarget = primaryDomainInfo.target
+      const redirectionDomainList = dnsInfo.domainAssociation?.subDomains?.filter(
+        (subDomain) => subDomain.subDomainSetting?.prefix
+      )
+
+      /**
+       * Amplify only stores the prefix.
+       * ie: if I wanted to have a www.blah.gov.sg -> giberish.cloudfront.net,
+       * amplify will store the prefix as "www". To get the entire redirectionDomainSource,
+       * I would have to add the prefix ("www") with the primary domain (blah.gov.sg)
+       */
+      const redirectionDomainSource = `${redirectionDomainList?.[0].subDomainSetting?.prefix}.${primaryDomain}`
+
+      const userId = agency.id
+      const newLaunchParams = {
+        userId,
+        siteId,
+        primaryDomainSource: primaryDomain,
+        primaryDomainTarget,
+        domainValidationSource,
+        domainValidationTarget,
+        redirectionDomainSource,
+      }
+
+      // Create launches records table
+      const launchesRecord = await this.launchesService.create(newLaunchParams)
+      logger.info(`Created launch record in database:  ${launchesRecord}`)
+
+      const message: MessageBody = {
+        repoName,
+        appId,
+        primaryDomainSource: primaryDomain,
+        primaryDomainTarget,
+        domainValidationSource,
+        domainValidationTarget,
+      }
+
+      if (redirectionDomainSource) {
+        message.redirectionDomain = [
+          {
+            source: redirectionDomainSource,
+            target: primaryDomainTarget,
+          },
+        ]
+      }
+
+      this.queueService.sendMessage(message)
+    } catch (error) {
+      logger.error(`Failed to created '${repoName}' site on Isomer: ${error}`)
+      throw error
+    }
+    return null
```

Comment:
addressed here d1f6dec using a diff solution

Reply:
actually the intention of returning `void` is different from `null`! the `Ok<T>` here means that we can unwrap and consume `T`, which doesn't seem to be the intention here - we only ever check if it `isOk` but after doing so, we don't consume `T` (here, `null`), which is why i suggested changing it to `void` (concrete type of `Ok<void>`), so that even if consumers were to unwrap, they still won't be able to do anything with the type.

Reply:
woah, seems like [neverthrow does not allow](https://github.com/supermacro/neverthrow/issues/97) for this ._.  
Given this limitation within neverthrow, I propose to keep this as it is for now?

Reply:
oh dang - since you're using neverthrow here, then yea lets keep it